express-notes-api/delete.js

```diff
@@ -0,0 +1,23 @@
+module.exports = function deleteEntry(app, data, path, fs, writeForDelete) {
+  app.delete('/api/notes/:id', (req, res) => {
+    const id = req.params.id;
+    const selectedNote = data.notes[id];
+    delete data.notes[id];
+    if (Number(id) < 0 || isNaN(id)) {
+      const negativeId = { error: 'id must be a positive integer' };
+      res.status(400).json(negativeId);
+      return;
+    } else if (selectedNote === undefined) {
```

Comment:
Since you return above, the `else` here is redundant. (Though, of course, the `if` is necessary.)